pkg/controller/plan/builder/vsphere/builder.go

```diff
@@ -148,38 +260,80 @@ func (r *Builder) host(hostID string) (host *vsphere.Host, err error) {
 
 //
 // Build the VMIO ResourceMapping CR.
-func (r *Builder) Mapping(mp *plan.Map, object *vmio.VirtualMachineImportSpec) (err error) {
+func (r *Builder) mapping(in *plan.Map, vm *model.VM) (out *vmio.VmwareMappings, err error) {
 	netMap := []vmio.NetworkResourceMappingItem{}
 	dsMap := []vmio.StorageResourceMappingItem{}
-	for i := range mp.Networks {
-		network := &mp.Networks[i]
+	for _, mapped := range in.Networks {
+		ref := mapped.Source
+		network := &model.Network{}
+		fErr := r.Inventory.Find(network, ref)
+		if fErr != nil {
+			err = liberr.Wrap(fErr)
+			return
+		}
+		needed := false
+		for _, net := range vm.Networks {
+			if net.ID == network.ID {
+				needed = true
+				break
+			}
+		}
+		if !needed {
```

Comment:
Determine if the VM references the network and conditionally included in the VMIO map.  This is more efficient and necessary because a network/datastore for another VM may not be known to this ESX host.